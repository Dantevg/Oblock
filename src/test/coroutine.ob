var inner, outer

inner := (() => {
	print "    inner before"
	outer.yield() -- yield outer coroutine, transfer control to main thread
	print "    inner after"
}).co()

outer := (() => {
	print "  outer before"
	inner() -- resume inner coroutine
	print "  outer after"
}).co()

print "outside before"
outer() -- resume outer coroutine
print "outside between"
outer() -- resume inner coroutine (the one that yielded)
print "outside after"
