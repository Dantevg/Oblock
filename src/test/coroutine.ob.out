outside before
  outer before
    inner before
outside between
    inner after
  outer after
outside after
